### PR TITLE
Add Section on Type Conversion; fixes #472

### LINKED ISSUE #472: Add section on type conversion to rules-dsl.md
Suggested by @rkoshak in comments related to pull request #469, a section on type conversion needs to be added to the file configurations/rules-dsl.md.
Content of this section will be based on forum contribution https://community.openhab.org/t/type-conversions/32684 (also by @rkoshak). I am planning to develop a draft that can be discussed in this thread.

### PR DESCRIPTION
**Pull request by LightIsLife**
Propose to add a section on type conversion to the documentation on Rules to address #472.
Proposed text originates from @rkoshak's forum contribution discussed here: https://community.openhab.org/t/type-conversions/32684; pertinent discussion in #469 also referenced here for completeness
Signed-off-by: Markus Lipp <lipp.markus@gmail.com> (github: LightIsLife)

### CODE REVIEW

**Comment by ThomDietrich** [comment]
You are eager, I like it!

In my personal opinion I thought the article is not ready yet. There is still a lot of information missing and I'd like to straighten out the code examples. However as the topic didn't receive much comments in the community (surprisingly) we can do that there in the PR...

**Comment by LightIsLife** [comment]
Thanks. Just trying to be helpful, there are some topics that I know (from experience) can be confusing for the less initiated...this is one of them. I will gladly take pointers do some more research, etc; I think we covered all types and tried to include all feedback that appeared in the forum, but as you said there was not a lot.

**Comment by LightIsLife** [comment]
@ThomDietrich @rkoshak Hi guys, I am a little stuck as I am not sure I am in a position to evaluate what needs to be changed without some more feedback/comments. However, if plans changed and this section is not desirable anymore, I will just close this PR. No harm done.

**Comment by rkoshak** [comment]
Don't let perfect be the enemy of the good. I say close the PR out so what we have goes live. Then we can open new PRs to cover the missing info. Sound like a plan?

**Comment by LightIsLife** [comment]
Works for me, no problem. I appreciate your support! Am I right to assume that your call for action (closing the PR) was addressed to the maintainers, as I believe if I did that it would get cancelled. Let me know if I misunderstood.

**Comment by rkoshak** [comment]
Right, someone needs to merge the changes and then close the PR.

**Comment by ThomDietrich** [comment]
Hey guys,
I was away for a few days. @rkoshak you are right, we could merge the PR right away. @LightIsLife no need to be overly humble, we all equals here ;) The section is very important and needs to be merged!

I didn't have the time yet to do a full review but I think we should invest some more time into this new section of the documentation. Now you @LightIsLife said you don't know where to continue. If you prefer, we can merge right away and be done with it or open another PR. I think the article would greatly benefit from:

* common structure: See Location Item. All Item sections should have these clear code examples. If you check others, the examples are often a bit random.
* Switch vs. Dimmer: I think it would be wise to write a Switch text, even if it is a copy of the Dimmer text.
* Markdown: In Kramdown, our Markdown engine used for docs.openhab.org, you need to add empty lines before and after blocks (e.g. code blocks). I found a few places where there is none.

What do you think?

**Comment by LightIsLife** [comment]
@ThomDietrich thanks much, that helps. Give me a couple of days and I can likely address most of it in this PR.

**Comment by LightIsLife** [comment]
@ThomDietrich @rkoshak hope I addressed all needs w/o adding more issues; as always, your review and guidance will be appreciated

File: configuration/rules-dsl.md [hunk 1]
```diff
@@ -450,9 +452,6 @@ val location = new PointType(new DecimalType(50.12345), new DecimalType(10.12345
 // Creation from String; ATTENTION: do not add space after comma
 val PointType home = new PointType("12.121212,123.123123")
 
-// Saving to an Item
-Device_Coordinates.postUpdate(location)
```
Comment (ThomDietrich) [inline configuration/rules-dsl.md]: Why did you remove this example?

Comment (LightIsLife) [reply configuration/rules-dsl.md]: I removed all the postUpdate and sendCommand examples as this is handled in the section above in more details. No strong feelings either way, it appeared redundant and superfluous to me at this point, but happy to put it back in, as it won't do harm either

Comment (ThomDietrich) [reply configuration/rules-dsl.md]: Ah. I see. Nope sounds good!

**Comment by LightIsLife** [comment]
@ThomDietrich i see that this has now confilcts, which had not been there a couple of days ago. Anything i need to do? I did not change anything....so not sure or lost track what happened

**Comment by ThomDietrich** [comment]
Nope, that happened because I did merge other changes that were in conflict with the ones in this PR.
Is the plan still to wait for the changes suggested above? We can of course also merge now and open a new PR with improvements. That would be okay with me, as long as the comments do not get lost ;)

**Comment by LightIsLife** [comment]
Thanks much. Honestly, not sure what comments still need to be addressed, at least i tried to address all review and other comments. Let me know if you still see anything that needs fixin'

**Comment by ThomDietrich** [comment]
Ah wait, seeing the intermediate commit just now. All set 👍 Here we go.

Congrats on the merged PR and thanks for your help making the documentation great!
